livekit-rtc/livekit/rtc/participant.py

```diff
@@ -236,6 +245,204 @@ async def publish_transcription(self, transcription: Transcription) -> None:
+        try:
+            resp = FfiClient.instance.request(req)
+            cb = await queue.wait_for(
+                lambda e: (e.perform_rpc.async_id == resp.perform_rpc.async_id)
+            )
+        finally:
+            FfiClient.instance.queue.unsubscribe(queue)
+
+        if cb.perform_rpc.HasField("error"):
+            raise RpcError.from_proto(cb.perform_rpc.error)
+
+        return cb.perform_rpc.payload
+
+    def register_rpc_method(
+        self,
+        method: str,
+        handler: Callable[[RpcInvocationData], Union[Awaitable[str], str]],
+    ) -> None:
+        """
+        Establishes the participant as a receiver for calls of the specified RPC method.
+        Will overwrite any existing callback for the same method.
+
+        Args:
+            method (str): The name of the indicated RPC method
+            handler (Callable): Will be invoked when an RPC request for this method is received
+
+        Returns:
+            None
+
+        Raises:
+            RpcError: On failure. Details in `message`.
+
+        Example:
+            async def greet_handler(data: RpcInvocationData) -> str:
+                print(f"Received greeting from {data.caller_identity}: {data.payload}")
+                return f"Hello, {data.caller_identity}!"
+
+            await room.local_participant.register_rpc_method('greet', greet_handler)
+
+        The handler should return a string or a coroutine that resolves to a string.
+
+        If unable to respond within `response_timeout`, the caller will hang up and receive an error on their side.
+
+        You may raise errors of type `RpcError` with a string `message` in the handler,
+        and they will be received on the caller's side with the message intact.
+        Other errors raised in your handler will not be transmitted as-is, and will instead arrive to the caller as `1500` ("Application Error").
+        """
+        self._rpc_handlers[method] = handler
+
+        req = proto_ffi.FfiRequest()
+        req.register_rpc_method.local_participant_handle = self._ffi_handle.handle
+        req.register_rpc_method.method = method
+
+        FfiClient.instance.request(req)
+
+    def rpc_method(self, method: str):
+        """
+        Decorator form of `register_rpc_method`
+
+        Args:
+            method (str): The name of the indicated RPC method
+
+        Example:
+            @local_participant.rpc_method("greet")
+            async def greet_handler(data: RpcInvocationData) -> str:
+                print(f"Received greeting from {data.caller_identity}: {data.payload}")
+                return f"Hello, {params.caller_identity}!"
+
+        See Also:
+            `register_rpc_method` for more details
+        """
+
+        def decorator(
+            handler: Callable[[RpcInvocationData], Union[Awaitable[str], str]],
+        ):
+            self.register_rpc_method(method, handler)
+            return handler
+
+        return decorator
```

Comment:
I think rpc_method and register_rpc_method should be the same method

Reply:
Example on how I do it for the EventEmitter: https://github.com/livekit/python-sdks/blob/5b78bcc816d9254be7b82421be2d18b80531335a/livekit-rtc/livekit/rtc/event_emitter.py#L119

Reply:
Yeah I was wondering if that's possible - I'd definitely prefer it that way. Thanks for the example!